internal/providers/github/selector_entity.go

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+
+	internalpb "github.com/stacklok/minder/internal/proto"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+// RepoToSelectorEntity converts a Repository to a SelectorEntity
+func (_ *GitHub) RepoToSelectorEntity(_ context.Context, r *minderv1.Repository) *internalpb.SelectorEntity {
+	fullName := fmt.Sprintf("%s/%s", r.GetOwner(), r.GetName())
+
+	return &internalpb.SelectorEntity{
+		EntityType: minderv1.Entity_ENTITY_REPOSITORIES,
+		Name:       fullName,
+		Entity: &internalpb.SelectorEntity_Repository{
+			Repository: &internalpb.SelectorRepository{
+				Name:      fullName,
+				IsFork:    proto.Bool(r.GetIsFork()),
+				IsPrivate: proto.Bool(r.GetIsPrivate()),
+			},
+		},
+	}
+}
+
+// ArtifactToSelectorEntity converts an Artifact to a SelectorEntity
+func (_ *GitHub) ArtifactToSelectorEntity(_ context.Context, a *minderv1.Artifact) *internalpb.SelectorEntity {
+	fullName := fmt.Sprintf("%s/%s", a.GetOwner(), a.GetName())
```

Comment:
Should this include `ghcr.io/`?  (I'm not sure)

Reply:
I would say that not the name attribute, but I agree with both you and @blkt (in [this comment](https://github.com/stacklok/minder/pull/3797#discussion_r1675647618)) that we should have a global name as well.

I think just based on what users would expect in the simplest case we could have:
 - `.name` - for repos this would be `orgname/reponame`
 - `.full_name` - this could be a clone name perhaps. For repos, `https://github.com/org/repo` for artifacts `registry_name/namespace/artifact_name`
 -  `.id` (help me come with better name :-)) - a minder-specific identifier. I liked @blkt's suggestion

btw note that as long as we expose the individual attributes the user can also combine them with logical operations in the CEL expression so we don't have to go overboard with providing too many one-off attributes.